picamraw/main_test.py

```diff
@@ -155,40 +155,64 @@ def test_integration(self):
 
 
 class TestUnpack10BitValues:
-    ''' Every 5 bytes contains the high 8-bits of 4 values followed by the low 2-bits of 4 values packed into 5th byte
-
-    Thus, for an array that contains these integers:
-        [1, 2, 3, 4, 5]
-    Represented as bytes (8-bits) in binary:
-        [0000001, 0000010, 0000011, 0000100, 0000101]
-    Unpack 5th byte as low 2-bits:
-        [000000100, 000001000, 000001101, 000010001]
-    Convert back to integers:
-        [4, 8, 13, 17]
-    '''
     def test_unpack_10bit_values(self):
+        # In the camera's RAW format, every 5 bytes in the source data contains the high 8-bits of
+        # 4 values followed by the low 2-bits of 4 values packed into 5th byte.
+        # Spec: https://linuxtv.org/downloads/v4l-dvb-apis-new/uapi/v4l/pixfmt-srggb10p.html
+        input_five_byte_set = [
+            # Note: In python, "0b" appears to the left of any binary value.
+            0b11111111,
+            0b10010010,
+            0b01001001,
+            0b00000000,
+            # Fifth byte is made up of the 2-bit sets that will be paired with the first 4 bytes
+            # In this case, from least to most significant, those 2-bit sets will be
+            # 0b00, 0b01, 0b10, 0b11
```

Comment:
I think you misordered these. Reading `0b11100100` right-to-left gets the pairs: `0b00`, `0b10`, `0b01`, `0b11`

Reply:
Ugh, or this is just still confusing. Not sure which it is.

Reply:
Ok, I think putting it this way would be less confusing to me:
byte 3: 0b11, byte 2: 0b10, byte 1: 0b01, byte 0: 0b00

While I (eventually) understood what you meant by bringing up big endian in this case, I think it ultimately added to rather than subtracted from the time it took me to understand.

Reply:
makes sense. no need to bring bigendianness into this. Clarified.